### PR TITLE
add waffle switch to disable tasks for maintenance

### PR DESCRIPTION
**Pull request by estherjsuh**
Task RED-2075

### CODE REVIEW

File: figures/tasks.py
Comment (OmarIthawi) [inline figures/tasks.py:399]:
```suggestion
        logger.info('%s: Disabled due to %s being active.', populate_daily_metrics.__name__,
```

File: figures/tasks.py
Comment (OmarIthawi) [inline figures/tasks.py:399]:
```suggestion
        logger.info('Figures pipeline is disabled due to %s being active.',
```

**Comment by OmarIthawi** [comment]
Thanks @estherjsuh! The tests are now failing because `django-waffle` wasn't added to the requirements:

```
    module = import_module(entry)
  File "/opt/hostedtoolcache/Python/2.7.18/x64/lib/python2.7/importlib/__init__.py", line 37, in import_module
    __import__(name)
ImportError: No module named waffle
ERROR: InvocationError for command /home/runner/work/figures/figures/.tox/py27-ginkgo/bin/pytest -c pytest-ginkgo.ini (exited with code 1)
```

Please fix that and it should be good to make the tests passes.

**Comment by OmarIthawi** [comment]
@estherjsuh I see that Juniper tests passes which is great. Please downgrade the `django-waffle` for both Ginkgo and Hawthorn.

File: tests/tasks/test_daily_tasks.py
Comment (OmarIthawi) [inline tests/tasks/test_daily_tasks.py:118]:
```suggestion
    populate_daily_metrics()
```

File: tests/tasks/test_daily_tasks.py
Comment (OmarIthawi) [inline tests/tasks/test_daily_tasks.py:128]:
```suggestion
    populate_daily_metrics()
```

**Review by OmarIthawi — COMMENTED** [review]
some fixes

File: tests/tasks/test_daily_tasks.py
Comment (OmarIthawi) [inline tests/tasks/test_daily_tasks.py:113]:
```suggestion
@pytest.mark.django_db
def test_disable_populate_daily_metrics(caplog):
```

**Review by OmarIthawi — APPROVED** [review]
That's great! Test passes and the code looks great to me. 

Thanks @estherjsuh! 

I'll wait for John to review it because we're both new to this code.

File: figures/tasks.py
Comment (johnbaldwin) [inline figures/tasks.py:56]: minor styling nit: 
> There's no blank line either before or after the docstring.

Ref: https://www.python.org/dev/peps/pep-0257/

File: figures/tasks.py
Comment (johnbaldwin) [inline figures/tasks.py:174]: minor styling nit: 
> There's no blank line either before or after the docstring.

Ref: https://www.python.org/dev/peps/pep-0257/

File: figures/tasks.py
Comment (johnbaldwin) [inline figures/tasks.py:393]: Minor nit: Please remove this comment line. The following conditional statement should be clear enough

File: tests/tasks/test_daily_tasks.py
Comment (johnbaldwin) [inline tests/tasks/test_daily_tasks.py:66]: From PEP8 (https://www.python.org/dev/peps/pep-0008/#imports)

> Imports should be grouped in the following order:
>
>    Standard library imports.
>    Related third party imports.
>    Local application/library specific imports.
>
> You should put a blank line between each group of imports.

So `import logging` should go above `import pytest` since `logging` is part of the standard library. The existing code (before your PR) does not have a blank line between the following, so while you reorder the imports, please add a blank line there.

```
from datetime import date
import pytest
```

Although I haven't seen it as part of PEP8, I generally order imports alphabetically within each group with the purpose to speed up visually scanning imports within a group. Admittedly, I'm not 100% consistent on this. But when I see it in my own code I've previously written, I try to fix it

 (but I'm not 100% consistent and don't want to be prescriptive here)

File: tests/tasks/test_daily_tasks.py
Comment (johnbaldwin) [inline tests/tasks/test_daily_tasks.py:77]: We don't need to change this.

**Review by johnbaldwin — CHANGES_REQUESTED** [review]
Hey @estherjsuh Great work! All my inline comments are on styling. Minor things, but if you would address those please.

You will need to merge from master into this PR branch

Thanks!

**Comment by estherjsuh** [comment]
@johnbaldwin Thank you for the review! I made the suggested changes, and I will merge once you have a final pass at it.

**Review by johnbaldwin — APPROVED** [review]
Great work, @estherjsuh ! Ship it!

**Pull request merged**
